datasets.py

```diff
@@ -694,3 +694,146 @@ def from_path(self, file_path):
+        self.people_labels = []
+        
+
+        # Number of images from dataset that are female and male (optional, doesn't need to exist)
+        #Note: Needs to exist for gender analysis, is count for each gender image
+        self.num_gender_images = [0, 0]
+        count = 0
+        with open('Anno/list_attr_celeba.txt') as f:
+            
+            for line in f:
+                if count >= 2:
+                    stripped_line = line.strip()
+                    stripped_line = stripped_line.split()
+                    image_values = stripped_line[1:]
+                    gender = int(image_values[20])
+                    if gender > 0:
+                        self.num_gender_images[1] += 1
+                    else:
+                        self.num_gender_images[0] += 1
+                count += 1
+                
+        print("done with gender counting")
+        print(self.num_gender_images)
+        
+    def __getitem__(self, index):
+        image_id = self.image_ids[index]
+        file_path = self.img_folder + '/' + image_id
+        return self.from_path(file_path)
+
+    def __len__(self):
+        return len(self.image_ids)
+
+    def from_path(self, file_path):
+        image_id = os.path.basename(file_path)
+
+        image = Image.open(file_path).convert("RGB")
+        image = self.transform(image)
+        image_size = list(image.size())[1:]
+
+
+
+        country = None # optional
+
+        image_anns = []
+        #Note: This is a map of present labels: image_anns = [{‘label’: ‘mustache’}, {‘label’: ‘young’}] where the image has a 1 for mustache and young (it's present)
+
+        with open('Anno/list_attr_celeba.txt') as f:
+            
+            for line in f:
+                stripped_line = line.strip()
+                if image_id in stripped_line:
+                    stripped_line = stripped_line.split()
+                    image_values = stripped_line[1:]
+                    gender = int(image_values[20])
+                    for category in range(len(self.categories)):
+                        if int(image_values[category]) == 1:
+                            image_anns.append({'label':self.categories[category]})
+                    
+                    break
+        with open('Anno/list_bbox_celeba.txt') as bbox_f:
+            for line in bbox_f:
+                stripped_line = line.strip()
+                if image_id in stripped_line:
+                    stripped_line = stripped_line.split()
+                    bbox = stripped_line[1:]
+                    
+                    #Note: bbox digits should be : x,y,width,height
+                    bbox_digits = [int(bbox[0]) / image_size[1], (int(bbox[0])+int(bbox[2])) / image_size[1], int(bbox[1]) / image_size[0], (int(bbox[1])+int(bbox[3])) / image_size[0]]
+                    break
+        
+
+        #Note: I think the default comment may not be correct, and females are 1 and males are 0
```

Comment:
good catch, this has been changed in the code base so this comment can be removed